### DISCUSSION TITLE
Stuck in RECONNECTING

### DISCUSSION THREAD

**Discussion by DrRuhe** [tokio-rs/console#360] [Q&A] [answered]
I tried setting up a project of mine with tokio-console. While the `app.rs` example worked, trying to connect to my project just gives me the following in tokio-console (It has a blank screen until I press a button, e.g. pressing a): 
![image](https://user-images.githubusercontent.com/29500714/177840508-50090b05-08f4-4f65-979a-f0484da84086.png)
My dependencies currently look like this:
```
tokio = { version = "1.18.2", features = ["full","tracing"] }
tracing = "0.1.35"
console-subscriber= "0.1.6"
actix-web = "4.0.1"

// should be irrellevant to problem.
entity = { path = "entity" }
migration = { path = "migration" }
chrono = "0.4.19"
serde = { version = "1.0.137", features = ["derive"] }
uuid = { version = "1.0.0", features = ["v4"] }
rand = "0.8.5"
itertools = "0.10.3"
sha2 = "0.10.2"
radix_trie = "0.2.1"
```
I activated tokio-console with the `console_subscriber::init();` directly as the first thing in main.
I have no other tracing-related setup (No configured global subscriber, etc.). I only have some async functions annotated with the `instrument` proc macro.

I am sure that I reach the application under 127.0.0.1:6669 since curl gives a different error than when trying to connect to an unused port.

Do you have an idea what the cause for this might be?

**Comment by grahamking** [comment, marked as answer]
Look in the logs `/tmp/tokio-console/logs/` to see if it says something like "OutOfRange, message: Error, message length too large..". 

If yes try setting the [retention](https://docs.rs/console-subscriber/latest/console_subscriber/struct.Builder.html#method.retention). The default is 1h.

**Reply by DrRuhe** [reply]
I can't test this anymore, given that it has been over a year. Thanks for the suggestion though, I am marking this as resolved. I think it is useful to track a better error message via an Issue so that other users don't run into the same.

**Comment by sukhmel** [comment]
@grahamking , I'm experiencing the same behaviour of `tokio-console`, there is no directory `/tmp/tokio-console` not on host with a service, not on the machine running `tokio-console`. Setting retention with environment variable `TOKIO_CONSOLE_RETENTION` to be 300 did nothing to the problem.

The problem is that I experience this with a real service that is a mess so I can't reduce it to a minimal example, and the reason to do that was because I expected there to be issues with tasks, so maybe it affects the console?

Edit: looks like I broke a build and it had tokio-console disabled, will check if that's the case and update accordingly 

Edit 2: there were two issues, one is tokio-console was not enabled, one is that network policies seem to prevent connecting by IP, but it worked with port-forwarding
